### PR TITLE
Upgrade TypeScript compiler to 1.7

### PR DESCRIPTION
**Pull request by Zenorbi**
Copied the files from typescript 1.7.3, modified typescriptServices.d.ts to include the extra `export = ts` and modified index.ts to comply to the new `readConfigFile` whereby the function expects an extra `readFile`.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -13,7 +14,7 @@ export function create(configOrName: builder.IConfiguration|string, verbose?: bo
     var config: builder.IConfiguration;
 
     if (typeof configOrName === 'string') {
-        var parsed = readConfigFile(configOrName);
+        var parsed = readConfigFile(configOrName, (path) => readFileSync(path, undefined));
         if (parsed.error) {
```
Comment (jrieken) [inline src/index.ts]: isn't `readConfigFile(configOrName, readFileSync)` enough?

**Comment by Zenorbi** [comment]
It should be, but node.d.ts does not allow that:
```
export function readFileSync(filename: string, encoding: string): string;
export function readFileSync(filename: string, options: { encoding: string; flag?: string; }): string;
export function readFileSync(filename: string, options?: { flag?: string; }): Buffer;
```
And none matches `(filename:string) => string`.

**Comment by jrieken** [comment]
ah, misses a ?

**Comment by jrieken** [comment]
I will merge and test. If things are happy I will bump the version and publish. Thanks

**Pull request merged**